### PR TITLE
Add SELECT 1 check to Exists/CanConnect checks

### LINKED ISSUE #18330: DatabaseFacade.CanConnect doesn't actually check anything
DatabaseFacade.CanConnect is supposed to determine whether the database is alive and can be connected to. The default implementation in RelationalDatabaseCreator calls Exists, whose implementation in SqlServer (and Sqlite and Npgsql) is to simply open and close a connection. As far as I'm aware, assuming connection pooling is enabled this doesn't actually do any sort of communication to the database. We should consider implementing CanConnect by doing a raw `SELECT 1` roundtrip, or possibly opening and closing a non-pooled connection.

Note that [ASP.NET's EF health probe](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-3.0#database-probe) is built over this API.

### PR DESCRIPTION
**Pull request by roji**
Fixes #18330

Note that the connection.Open operation has passes `errorsExpected: true`, so that connection errors are logged as debug. We can do this for the `SELECT 1` command as well, but I think the main idea here is to squash "does not exist" errors specifically, which really aren't an error. Other types of errors (e.g. connectivity) still seem like they should log an error, since it's not a matter of the database not being created. But if someone thinks otherwise I can add `errorsExpected` to command failure logging.

### CODE REVIEW

**Comment by ErikEJ** [comment]
Why the additional logging?

**Comment by roji** [comment]
@ErikEJ what do you mean?

**Comment by ErikEJ** [comment]
The call to .ExecuteNonQuery - but maybe it always is logged?

**Comment by roji** [comment]
Yeah, the call is logged like any other command execution (assuming you've enabled it etc.). Do you see an issue here?

We can specifically exempt logging this command, but I'm not sure whether that would be the right thing to do (see also comment in the OP).

File: src/EFCore.SqlServer/Storage/Internal/SqlServerDatabaseCreator.cs [hunk 1]
```diff
@@ -185,12 +185,18 @@ private bool Exists(bool retryOnNotExists)
...
+
+                            _rawSqlCommandBuilder
+                                .Build("SELECT 1")
+                                .ExecuteNonQuery(new RelationalCommandParameterObject(
+                                    _connection,
+                                    null,
+                                    Dependencies.CurrentContext.Context,
+                                    Dependencies.CommandLogger));
+
+                            _connection.Close();
```
Comment (AndriySvyryd) [inline src/EFCore.SqlServer/Storage/Internal/SqlServerDatabaseCreator.cs]: Also call `Close` in the `catch` block if an exception is thrown before this.

Comment (roji) [reply src/EFCore.SqlServer/Storage/Internal/SqlServerDatabaseCreator.cs]: Thanks, my bad.

**Pull request merged**